tools/oesign/main.c

```diff
@@ -743,23 +828,39 @@ int sign_parser(int argc, const char* argv[])
         }
     } while (1);
 
-    if (enclave == NULL)
+    if (conffile == NULL)
     {
-        Err("Enclave image flag is missing");
+        Err("Config file flag is missing");
         ret = 1;
     }
-    if (conffile == NULL)
+
+#if HAS_ENGINE_SUPPORT
+    if (keyfile && engine_id)
     {
-        Err("Config file flag is missing");
+        Err("only one of keyfile and engine flags are allowed");
+        ret = 1;
+    }
+    if (keyfile == NULL && engine_id == NULL)
```

Comment:
Additional error cases to trap:
- `if (engine_id != NULL && key_id == NULL)` 
- `if (engine_id == NULL && (key_id != NULL || engine_load_path != NULL))`

It's better to trap these up front where useful messages can be displayed rather than letting them fall through to the methods that expect these conditions but only return `OE_INVALID_PARAMETER`

Reply:
done